math/calculator.cpp

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2020 @author David Leal
+ * @file
+ *
+ * Basic calculator: sum, subtract, multiply, and divide.
+ */
+
+#include <iostream>
+#include <cstdlib>
+
+/**
+ * Same as the number typed by the user, but it's saved
```

Comment:
based on the description, even after using the structure, the number will be overwritten. could simply use a variable inside the main function and use pointers to it. Thus avoiding memory allocations altogether and simplifying the program.

Reply:
Have you tested the program? It works perfectly for me.
AFAIK, number is not overwritten.

Reply:
unless, I misunderstood,
lines 95, 103, 129, etc each of them updates the value in the node - i.e., the value is overwritten.

Reply:
> Have you tested the program? It works perfectly for me.
> AFAIK, number is not overwritten.

The program would work; but the comment is incorrect and thus rendering the use of node unnecessary.

Reply:
How may this be fixed, then?

Reply:
1. remove struct between line 16-18
2. instead of Node* on line 24 and line 62, replace with `float *new_val`
3. replace all instances of `new_node->datatype` with `*new_val`
4. replace line 34 with `float global_var`
5. replace line 35 with `request_data(&global_var)`
6. remove line 41

Reply:
Should be done now.